django/db/backends/base/base.py

```diff
@@ -187,22 +191,41 @@ def get_database_version(self):
             "method."
         )
 
-    def check_database_version_supported(self):
-        """
-        Raise an error if the database version isn't supported by this
-        version of Django.
-        """
+    async def aget_database_version(self):
+        """Return a tuple of the database's version."""
+        raise NotSupportedError(
+            "subclasses of BaseDatabaseWrapper may require a aget_database_version() "
+            "method."
+        )
```

Comment:
Should this use `sync_to_async` to call the synchronous version instead of failing entirely?

Reply:
I was worried about the performance hit, since this gets called every time a new connection is initiated, which would happen pretty often when using  `async with new_connection:`

Reply:
Would `async with new_connection` make sense for non-async backends? In other words, how would `async with new_connection` work for SQLite right now?

Reply:
I think in those cases we should throw an Exception, maybe `NotSupportedError`?

Reply:
I've added a guard in f7f86c303d5f4bb8b37c6dbcb208a298bcd521d2